src/vs/workbench/contrib/debug/browser/loadedScriptsPickerUtils.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as nls from 'vs/nls';
+import { matchesFuzzy } from 'vs/base/common/filters';
+import { Source } from 'vs/workbench/contrib/debug/common/debugSource';
+import { Codicon } from 'vs/base/common/codicons';
+import { IQuickPick, IQuickPickItem, IQuickPickSeparator } from 'vs/platform/quickinput/common/quickInput';
+import { IDebugSession, State } from 'vs/workbench/contrib/debug/common/debug';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+
+export interface IPickerLoadedScriptItem extends IQuickPickItem {
+	accept(): void;
+}
+
+// this function takes a regular quickpick and makes one for loaded scripts that has persistent headers
+// e.g. when some picks are filtered out, the ones that are visible still have its header
+export async function configureLoadedScriptMenu(quickPick: IQuickPick<IPickerLoadedScriptItem>, sessions: IDebugSession[], editorService: IEditorService) {
+	quickPick.matchOnLabel = quickPick.matchOnDescription = quickPick.matchOnDetail = quickPick.sortByLabel = false;
+	quickPick.placeholder = nls.localize('moveFocusedView.selectView', "search for a loaded script");
+	quickPick.items = await _getPicks(quickPick.value, sessions, editorService);
+	quickPick.onDidChangeValue(async () => {
```

Comment:
The quickPick and event listeners on it need to be disposed, see other usages like quickAccess.ts for an example.